Add shared agentic issue triage template for all Azure SDK repos (#15325)
Azure/azure-sdk-tools: tools/agentic-triage/issue-triage-template.md

#   2. Replace all {{PLACEHOLDERS}} with language-specific values (search for "{{")
#   3. Run: gh aw compile .github/workflows/issue-triage.md
#   4. Commit both issue-triage.md and issue-triage.lock.yml
#   5. Add COPILOT_GITHUB_TOKEN repository secret
#
# LANGUAGE-SPECIFIC VALUES TO REPLACE:
#   {{LANGUAGE}}              → Java | Python | JavaScript | .NET | Rust
#   {{REPO_SLUG}}             → azure/azure-sdk-for-java (lowercase)
#   {{PACKAGE_DESCRIPTION}}   → e.g. "Maven artifacts beginning with com.azure"
#   {{PACKAGE_EXAMPLES}}      → e.g. "com.azure.cosmos, com.azure.storage.blob"
#   {{MGMT_PATTERN}}          → e.g. "Maven group com.azure.resourcemanager"
#   {{DOCS_URL}}              → e.g. "https://learn.microsoft.com/java/azure/"
#   {{API_DOCS_URL}}          → e.g. "https://azure.github.io/azure-sdk-for-java/"
#   {{EXTRA_TYPE_LABELS}}     → Additional type labels if needed (or remove block)
#   {{SEARCH_EXAMPLE_CLASS}}  → e.g. "BlobServiceClient" or "SecretAsyncClient"
#   {{SEARCH_EXAMPLE_ERROR}}  → e.g. "NullPointerException SecretAsyncClient"
#   {{SEARCH_EXAMPLE_MODULE}} → e.g. "computeIfAbsent processorMap"

> # ===========================================================================

Reviewer: Will an agent running the agentic workflow see this? Or will it only see the markdown text after the frontmatter?

Author: Markdown text after the frontmatter, this will be treated as YAML comments, so they're stripped out during compilation

3. Analyze issue content

   - Title and description
   - Type: bug report, feature request, question, documentation issue, etc.
   - Technical areas mentioned
   - Severity or priority indicators
   - User impact
   - Package names: {{PACKAGE_DESCRIPTION}} (e.g. {{PACKAGE_EXAMPLES}})
   - Service SDK directories under `/sdk/` (e.g. `sdk/cosmos`, `sdk/storage`, `sdk/keyvault`)
   - Changed files in linked pull requests
   - Stack traces, error messages, or exception types mentioned

<!-- CUSTOMIZATION: POST_ANALYSIS — add language-specific analysis steps here.
     Example (.NET): Check NuGet for package deprecation using web-fetch.
     Example (JS): Check npm for package deprecation using web-fetch. -->

4. Write notes, ideas, nudges, resource links, debugging strategies, and reproduction steps relevant to the issue

   - Reference relevant Azure SDK for {{LANGUAGE}} documentation: {{DOCS_URL}}
   - Reference relevant API docs: {{API_DOCS_URL}}
   - Link to relevant troubleshooting guides if the issue relates to a known service area (e.g. `sdk/<service>/azure-<service>/TROUBLESHOOTING.md`)

5. Select appropriate labels from available repo labels

>    These rules are aligned with the canonical `github-event-processor` in `azure-sdk-tools`.

Reviewer: Have we tested how the workflow does without this line? 
Curious if it's actually having the agentic workflow look at the code for the event processor and using that to confirm labeling process.

Author: yeah, before I introduced the github-event-processor terms in the workflow the agent was still able to fetch all labels and use them accurately, see here https://github.com/samvaity/azure-sdk-for-java/commit/824e0a4c58111422a9057a0238128b0239369f70

It's just a comment for humans. The agent doesn't fetch or read the event processor code — it follows the specific rules written out in the steps below that line

   - All issues should have a #ffeb77 colored type label
     - `Client` - client libraries not matching management package patterns
     - `Mgmt` - management libraries ({{MGMT_PATTERN}}) or mentions of ARM or Resource Manager

>      - `Service` - REST API or service behavior outside client SDK control

Reviewer: Any issue that is labeled with `Service` should be redirected. The customer should open up a support ticket with the actual service. 

I'm also having trouble finding any instances where a bot successfully labeled an issue with this label. This might be something to look at updating in our current workflow. Should this label be in this category, or should we just do a check on if the issue is an SDK issue or a service issue earlier and if it's a service issue comment letting the user know they should open a support ticket and mark the issue as resolved (marking the issue as resolved allows the author to say it wasn't resolved if the issue was misidentified).

     {{EXTRA_TYPE_LABELS}}

>    - **Customer detection** (aligned with github-event-processor `InitialIssueTriage` rule):

Reviewer: Again, interest to see if comments pointing to the github event processor rules are actually affecting the agentic workflow's success.

Author: Nope, they are not. The workflow only takes reference to the actual rules spelled out below it

     - For external customers: add `customer-reported` and `question` labels
     - Note: `question` is added to ALL external customer issues by the event processor regardless of issue type — this is the existing behavior
   - If the issue is already assigned, do not apply `customer-reported`, `needs-triage`, or `needs-team-triage` labels
   - Add `EngSys` service label for issues with scripts, workflows, or pipelines under /eng but not /eng/common
   - Use labels from similar issues for #e99695 colored service labels
   - If pull requests are linked to similar issues, check those pull requests' file paths against matching patterns in /.github/CODEOWNERS
     - If matches are found, use the `PRLabel` value in a comment above those lines (e.g. `PRLabel: %KeyVault`) to find related `ServiceLabel`s (e.g. `ServiceLabel: %KeyVault`) grouped with `AzureSdkOwners` and `ServiceOwners`
     - Strip leading `@` from users and groups when assigning issues
     - Strip leading `%` from labels
     - Add #e99695 colored service labels from `ServiceLabel`
     - **Routing logic** (aligned with github-event-processor):

>        - If `Client` is applicable and there are `AzureSdkOwners` with valid repo permissions, and the issue is not already assigned: use `assign_to_user` to assign a random owner AND add `needs-team-attention`

Reviewer: Do we need to state that it should assign a random owner from the `AzureSdkOwners`? 
Has it ever accidentally pulled a github username from the `ServiceOwners` line?

Author: I was hesitant on this one. We haven't seen it pull random owners in the testing so far at least. But would you suggest we add a group of owners per repo to get assigned to in case this happens?

       - If NO `AzureSdkOwners` can be assigned but `ServiceOwners` exist: add `Service Attention` label (this is the fallback path)

>        - If the issue is already assigned: add `Service Attention` instead of re-assigning

Reviewer: Do we need to state on this line that this action should only happen if No `AzureSdkOwners` were listed as well?

     - Comment using this template when routing:

       ```markdown
       Thank you for your feedback. Tagging and routing to the team members best able to assist. cc {{ `AzureSdkOwners` each prefaced with `@` }}
       ```

     - If `Service` is applicable, add applicable labels and `needs-triage`, then exit
   - All issues should have a #e99695 colored service label describing the relevant service
   - **Triage label logic** (aligned with github-event-processor):
     - `needs-triage`: Apply when unable to predict ANY labels (cannot classify the issue at all)

>      - `needs-team-triage`: Apply when labels ARE predicted but no valid `AzureSdkOwners` can be assigned AND `Service Attention` is not used

Reviewer: This label isn't specific to just the Azure SDK team. It should still be applied if service labels were properly determined. 

It should be :
```suggestion
     - `needs-team-triage`: Apply when labels ARE predicted but no valid `AzureSdkOwners` or `ServiceOwner` can be assigned
```

>      - `needs-team-attention`: Apply when labels ARE predicted AND a valid `AzureSDKOwner` is assigned to the issue

Reviewer: The `needs-team-attention` label should be added if `ServiceOwners` were present. This label just means the next action is on the issue assignee and not the author. It's not specific to the Azure SDK team.

     - These three labels are mutually exclusive — only one should be applied

> 6. For bug-type issues, evaluate whether Copilot coding agent can handle the fix

Reviewer: Is there any benefit to braking out the evaluation of "is a bug issue a good issue for the coding agent to handle" into its own agentic workflow? 

It would trigger on an issue update event for validate that the update included the `bug` label being added.

Author: I see the benefits, but do have concerns about the agent redoing the research, the agent already has all the context (similar issues, PRs) so kept it inline to avoid the duplicate research. 
Once we see that the copilot assignment logic is getting more complex, breaking it into its own workflow triggered on issues: labeled makes sense. Could be a good v2 improvement

   - This step applies when ALL of the following conditions are met:

>      a. The issue is clearly a bug report (not a feature request, question, or service issue)

Reviewer: What does it mean for an issue to "clearly" be a bug report. We should define this.

     b. A similar past issue was found that was closed with a merged pull request
     c. The past fix was localized — the PR changed files in a single SDK package directory (e.g. only files under `sdk/cosmos/`)
     d. The current issue describes a similar or related problem in the same package area
     e. The issue has clear reproduction steps or a specific error/stack trace
   - If ALL conditions are met:

>      - Use `assign_to_user` to assign `copilot` to the issue

Reviewer: Do we want to jump right to auto assigning copilot? Or see how the workflow does at identifying candidates first?

We might need to think about how this works with service team owned libraries. Do they want this or will copilot opening these PRs for them to review create noise.

Author: Hmm, good point. In that case, we could start with a flag it in the analysis comment ('🤖 This
looks like a good candidate for Copilot') without actually assigning. That lets us validate the agent's judgment first.  Once we're confident in the candidate selection, we can flip to auto-assign.

Or we can even start with our own team's packages before expanding to service-team-owned libraries
